--- a/lib/ioctl/libdm-compat.h	2003-11-13 08:14:28.000000000 -0500
+++ b/lib/ioctl/libdm-compat.h	2003-12-09 11:51:53.000000000 -0500
@@ -31,7 +31,7 @@
 	int32_t open_count;	/* out */
 	uint32_t flags;		/* in/out */
 
-	__kernel_dev_t dev;	/* in/out */
+	__kernel_old_dev_t dev;	/* in/out */
 
 	char name[DM_NAME_LEN];	/* device name */
 	char uuid[DM_UUID_LEN];	/* unique identifier for
@@ -51,7 +51,7 @@
 struct dm_target_deps_v1 {
 	uint32_t count;
 
-	__kernel_dev_t dev[0];	/* out */
+	__kernel_old_dev_t dev[0];	/* out */
 };
 
 enum {
